store/copr/coprocessor_test.go

```diff
@@ -53,105 +53,252 @@ func TestBuildTasks(t *testing.T) {
-	taskEqual(t, tasks[1], regionIDs[1], "g", "n")
-	taskEqual(t, tasks[2], regionIDs[2], "n", "t")
-	taskEqual(t, tasks[3], regionIDs[3], "t", "x")
+	taskEqual(t, tasks[0], regionIDs[0], 0, "a", "g")
+	taskEqual(t, tasks[1], regionIDs[1], 0, "g", "n")
+	taskEqual(t, tasks[2], regionIDs[2], 0, "n", "t")
+	taskEqual(t, tasks[3], regionIDs[3], 0, "t", "x")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("a", "b", "b", "c"), req, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 1)
-	taskEqual(t, tasks[0], regionIDs[0], "a", "b", "b", "c")
+	taskEqual(t, tasks[0], regionIDs[0], 0, "a", "b", "b", "c")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("a", "b", "b", "c"), flashReq, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 1)
-	taskEqual(t, tasks[0], regionIDs[0], "a", "b", "b", "c")
+	taskEqual(t, tasks[0], regionIDs[0], 0, "a", "b", "b", "c")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("a", "b", "e", "f"), req, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 1)
-	taskEqual(t, tasks[0], regionIDs[0], "a", "b", "e", "f")
+	taskEqual(t, tasks[0], regionIDs[0], 0, "a", "b", "e", "f")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("a", "b", "e", "f"), flashReq, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 1)
-	taskEqual(t, tasks[0], regionIDs[0], "a", "b", "e", "f")
+	taskEqual(t, tasks[0], regionIDs[0], 0, "a", "b", "e", "f")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("g", "n", "o", "p"), req, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 2)
-	taskEqual(t, tasks[0], regionIDs[1], "g", "n")
-	taskEqual(t, tasks[1], regionIDs[2], "o", "p")
+	taskEqual(t, tasks[0], regionIDs[1], 0, "g", "n")
+	taskEqual(t, tasks[1], regionIDs[2], 0, "o", "p")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("g", "n", "o", "p"), flashReq, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 2)
-	taskEqual(t, tasks[0], regionIDs[1], "g", "n")
-	taskEqual(t, tasks[1], regionIDs[2], "o", "p")
+	taskEqual(t, tasks[0], regionIDs[1], 0, "g", "n")
+	taskEqual(t, tasks[1], regionIDs[2], 0, "o", "p")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("h", "k", "m", "p"), req, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 2)
-	taskEqual(t, tasks[0], regionIDs[1], "h", "k", "m", "n")
-	taskEqual(t, tasks[1], regionIDs[2], "n", "p")
+	taskEqual(t, tasks[0], regionIDs[1], 0, "h", "k", "m", "n")
+	taskEqual(t, tasks[1], regionIDs[2], 0, "n", "p")
 
 	tasks, err = buildCopTasks(bo, cache, buildCopRanges("h", "k", "m", "p"), flashReq, nil)
 	require.NoError(t, err)
 	require.Len(t, tasks, 2)
-	taskEqual(t, tasks[0], regionIDs[1], "h", "k", "m", "n")
-	taskEqual(t, tasks[1], regionIDs[2], "n", "p")
+	taskEqual(t, tasks[0], regionIDs[1], 0, "h", "k", "m", "n")
+	taskEqual(t, tasks[1], regionIDs[2], 0, "n", "p")
+}
+
+func TestBuildTasksByBuckets(t *testing.T) {
+	mockClient, cluster, pdClient, err := testutils.NewMockTiKV("", nil)
```

Comment:
Should we use the new mockstore(`NewMockStore`) instead?

Reply:
No, `NewMockStore` doesn't export region schedule functions.